server/portal/apps/search/tasks.py

```diff
@@ -14,7 +14,7 @@
 
 # Crawl and index agave files
 @shared_task(bind=True, max_retries=3, queue='indexing', retry_backoff=True, rate_limit="12/m")
-def agave_indexer(self, systemId, filePath='/', recurse=True, update_pems=False, ignore_hidden=True, reindex=False):
+def tapis_indexer(self, access_token, refresh_token, systemId, filePath='/', recurse=True, update_pems=False, ignore_hidden=True, reindex=False):
```

Comment:
I think this method should take a `client` object, similar to other Tapis utility functions. That way we can pass in the client retrieved from `service_account()` for indexing public/community data.

Reply:
@shayanaijaz If celery is unable to take a `client` object, could we look up the client via a `username` param instead?

Reply:
Yeah we can't pass in the client since the celery task only accepts JSON serializable data types. Passing in the `username` and instantiating the client using the Django user model could work. I'll give that a try. But we will have to think of another way to use the `service_account()` for indexing public/community data (maybe a boolean flag as an argument?)